extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/VertxHttpRecorder.java

```diff
@@ -997,51 +998,51 @@ private static void setHttpServerTiming(boolean httpDisabled, HttpServerOptions
         int socketCount = 0;
 
         if (!httpDisabled && httpServerOptions != null) {
-            serverListeningMessage.append(String.format(
-                    "http://%s:%s", getDeveloperFriendlyHostName(httpServerOptions), actualHttpPort));
+            appendListeningMessage(serverListeningMessage, "http", httpServerOptions.getHost(), actualHttpPort);
             socketCount++;
         }
 
         if (sslConfig != null) {
             if (socketCount > 0) {
                 serverListeningMessage.append(" and ");
             }
-            serverListeningMessage
-                    .append(String.format("https://%s:%s", getDeveloperFriendlyHostName(sslConfig), actualHttpsPort));
+            appendListeningMessage(serverListeningMessage, "https", httpServerOptions.getHost(), actualHttpsPort);
             socketCount++;
         }
 
         if (domainSocketOptions != null) {
             if (socketCount > 0) {
                 serverListeningMessage.append(" and ");
             }
-            serverListeningMessage.append(String.format("unix:%s", getDeveloperFriendlyHostName(domainSocketOptions)));
+            serverListeningMessage.append(String.format("unix:%s", domainSocketOptions.getHost()));
```

Comment:
Domain sockets don't understand the any address concept, so we don't need special handling here.